[SecuritySolution] Sourcerer UI

angorayc opened the pull request:
## Summary

<img width="1663" alt="Screenshot 2021-10-12 at 15 39 02" src="https://user-images.githubusercontent.com/6295984/136977430-7679e4cd-3aec-4f47-a858-a4f4e8a3eb6e.png">
<img width="1661" alt="Screenshot 2021-10-12 at 15 39 41" src="https://user-images.githubusercontent.com/6295984/136977438-c9d19144-bf98-4728-8930-def592225276.png">
<img width="1666" alt="Screenshot 2021-10-12 at 15 39 19" src="https://user-images.githubusercontent.com/6295984/136977441-c534599f-1684-45d2-b20c-2902a8ef5e81.png">

Checklist

- [x]  Any text added follows EUI's writing guidelines, uses sentence case text and includes i18n support
- [x]  Unit or functional tests were updated or added to match the most common scenarios

angorayc committed:
styling for sourcerer

angorayc committed:
move data sourcerer to global header

angorayc committed:
add modified badge

angorayc committed:
alerts default selected options

angorayc committed:
timeline sourcerer

angorayc committed:
security data view badge in kibana settings

angorayc committed:
copy change

angorayc committed:
styling

angorayc committed:
unit tests

angorayc committed:
remove unused i18n keys

angorayc committed:
hide sourcerer on certain pages

angorayc committed:
update snapshot

angorayc committed:
tests

angorayc committed:
unit tests

angorayc committed:
add usePickIndexPatterns hook

angorayc committed:
add helpers

angorayc committed:
unit test

angorayc committed:
close advance option by default

angorayc commented:
![sourcerer-ui](https://user-images.githubusercontent.com/6295984/137721105-6e7d23a2-5f2c-4f3d-85b6-ccd1232f4ccd.gif)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161970/)
* Commit: 3ca149d62aa6015ebdccba6a04aee8dc16a707a5
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-114805/3ca149d62aa6015ebdccba6a04aee8dc16a707a5)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-securitySolutionCypressChrome](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161970/execution/node/534/log)
* [Execute xpack-securitySolutionCypressChrome](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161970/execution/node/1096/log)

### History
* :broken_heart: [Build #161014](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161014/) failed a467cae9fd7e5a78b13caf4cb7d4c735d10e25c6
* :broken_heart: [Build #160853](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/160853/) failed 8a5f8ce77602356b4aad645fe3a09fd3e33b56ad
* :broken_heart: [Build #160493](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/160493/) failed d575aa3014c7dd9b2cf13b657e92345153f1dabf
* :broken_heart: [Build #160463](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/160463/) failed 967b7205a48f6f62694919f4fb834d41fc4f5186

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/160463/","number":"160463","commit":"967b7205a48f6f62694919f4fb834d41fc4f5186"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/160493/","number":"160493","commit":"d575aa3014c7dd9b2cf13b657e92345153f1dabf"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/160853/","number":"160853","commit":"8a5f8ce77602356b4aad645fe3a09fd3e33b56ad"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161014/","number":"161014","commit":"a467cae9fd7e5a78b13caf4cb7d4c735d10e25c6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161970/","number":"161970","commit":"3ca149d62aa6015ebdccba6a04aee8dc16a707a5"}]}
PIPELINE-->

angorayc committed:
hide global sourcerer when timeline is open

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/163153/)
* Commit: 02c559ece9a527b7ad1a13c2ad77542ac6229d0c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-114805/02c559ece9a527b7ad1a13c2ad77542ac6229d0c)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroupDocker](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/163153/execution/node/720/log)
* [Execute xpack-kibana-ciGroupDocker](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/163153/execution/node/760/log)

### History
* :broken_heart: [Build #161970](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161970/) failed 3ca149d62aa6015ebdccba6a04aee8dc16a707a5
* :broken_heart: [Build #161014](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161014/) failed a467cae9fd7e5a78b13caf4cb7d4c735d10e25c6
* :broken_heart: [Build #160853](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/160853/) failed 8a5f8ce77602356b4aad645fe3a09fd3e33b56ad
* :broken_heart: [Build #160493](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/160493/) failed d575aa3014c7dd9b2cf13b657e92345153f1dabf
* :broken_heart: [Build #160463](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/160463/) failed 967b7205a48f6f62694919f4fb834d41fc4f5186

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/160463/","number":"160463","commit":"967b7205a48f6f62694919f4fb834d41fc4f5186"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/160493/","number":"160493","commit":"d575aa3014c7dd9b2cf13b657e92345153f1dabf"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/160853/","number":"160853","commit":"8a5f8ce77602356b4aad645fe3a09fd3e33b56ad"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161014/","number":"161014","commit":"a467cae9fd7e5a78b13caf4cb7d4c735d10e25c6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161970/","number":"161970","commit":"3ca149d62aa6015ebdccba6a04aee8dc16a707a5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/163153/","number":"163153","commit":"02c559ece9a527b7ad1a13c2ad77542ac6229d0c"}]}
PIPELINE-->

angorayc committed:
Merge remote-tracking branch 'upstream/sourcerer_kip_as' into sourcerer-ui

angorayc committed:
fix up

angorayc committed:
unit test

kibanamachine committed:
Merge branch 'sourcerer_kip_as' into sourcerer-ui

angorayc committed:
Merge remote-tracking branch 'upstream/sourcerer_kip_as' into sourcerer-ui

angorayc commented:
@elasticmachine merge upstream

kibanamachine commented:
merge conflict between base and head

elasticmachine commented:
Pinging @elastic/security-threat-hunting (Team:Threat Hunting)

elasticmachine commented:
Pinging @elastic/security-solution (Team: SecuritySolution)

angorayc committed:
Merge branch 'sourcerer_kip_as' of github.com:elastic/kibana into sourcerer-ui

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/3196)
* Commit: ff8fa4c186859a0c8704ce328fc3e788e08ba5cf
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-114805/ff8fa4c186859a0c8704ce328fc3e788e08ba5cf)

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/3196#a5944c35-d6df-464f-8f2e-b0de23ecdfa5)

### History
* :broken_heart: [Build #1743](https://buildkite.com/elastic/kibana-pull-request/builds/1743) failed 09e70ee0ec2b631489f9acce65d2a84a0cb703aa
* :broken_heart: [Build #878](https://buildkite.com/elastic/kibana-pull-request/builds/878) failed d83cc97980995553dcd4b23b97fd40dc7d715066
* :broken_heart: [Build #835](https://buildkite.com/elastic/kibana-pull-request/builds/835) failed 20a9f35b98143566e13c41a6ec83ca17f9d339ab

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/3196","number":3196,"commit":"ff8fa4c186859a0c8704ce328fc3e788e08ba5cf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/1743","number":1743,"commit":"09e70ee0ec2b631489f9acce65d2a84a0cb703aa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/878","number":878,"commit":"d83cc97980995553dcd4b23b97fd40dc7d715066"},{"buildStatus":{"state":"running","success":false,"hasRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/835","number":835,"commit":"20a9f35b98143566e13c41a6ec83ca17f9d339ab"}],"number":3196}
buildkite-pr-comment-->

machadoum committed:
Fix eslint issue

machadoum commented on x-pack/plugins/security_solution/public/common/components/sourcerer/index.test.tsx:
```diff
@@ -64,12 +69,52 @@ describe('Sourcerer component', () => {
+  afterAll(() => {
+    (useRouteSpy as jest.Mock).mockImplementation(() => [
+      { pageName: SecurityPageName.overview, detailName: undefined },
+    ]);
```
Why is it necessary? I think that setting the mock implementation inside `beforeAll` should be enough. 🤔

angorayc replied:
yeah, we don't need this.

machadoum commented on x-pack/plugins/security_solution/public/common/components/sourcerer/index.test.tsx:
```diff
@@ -522,3 +567,167 @@ describe('Sourcerer component', () => {
   });
 });
+
+describe('sourcer on alerts page or rules details page', () => {
```
typo? `sourcerer `

angorayc replied:
Thank you!

machadoum commented on x-pack/plugins/security_solution/public/app/home/global_header/index.test.tsx:
```diff
@@ -0,0 +1,113 @@
+ * 2.0.
+ */
+import React from 'react';
+import { mount } from 'enzyme';
```
I think that we should only create new test files using `react test utils`. (not sure)

machadoum commented on x-pack/plugins/security_solution/public/common/components/sourcerer/use_pick_index_patterns.tsx:
```diff
@@ -0,0 +1,164 @@
+  setIndexPatternsByDataView: (newSelectedDataViewId: string) => void;
+}
+
+export const usePickIndexPatterns = ({
```
Nice cleanup! 👏

machadoum reviewed (approved):
Hi @angorayc,

I left some nit comments. Nothing worrisome. 
The feature works smoothly, great job!

Could you take a look at this scrolling issue? https://github.com/elastic/kibana/pull/114805#issuecomment-956303336

angorayc committed:
fix tests

angorayc committed:
Merge branch 'sourcerer-ui' of github.com:angorayc/kibana into sourcerer-ui

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/3278)
* Commit: b6a26cc81faa7efafc136dd1c49eab512630a330
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-114805/b6a26cc81faa7efafc136dd1c49eab512630a330)

### Failed CI Steps
* [Security Solution Tests](https://buildkite.com/elastic/kibana-pull-request/builds/3278#5c68dc7b-a215-4379-b59a-892f96d48ef5)
* [Security Solution Tests](https://buildkite.com/elastic/kibana-pull-request/builds/3278#2dbab452-d8d7-4356-a3d2-5908f43081bb)

### History
* :broken_heart: [Build #3196](https://buildkite.com/elastic/kibana-pull-request/builds/3196) failed ff8fa4c186859a0c8704ce328fc3e788e08ba5cf
* :broken_heart: [Build #1743](https://buildkite.com/elastic/kibana-pull-request/builds/1743) failed 09e70ee0ec2b631489f9acce65d2a84a0cb703aa
* :broken_heart: [Build #878](https://buildkite.com/elastic/kibana-pull-request/builds/878) failed d83cc97980995553dcd4b23b97fd40dc7d715066
* :broken_heart: [Build #835](https://buildkite.com/elastic/kibana-pull-request/builds/835) failed 20a9f35b98143566e13c41a6ec83ca17f9d339ab

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/3278","number":3278,"commit":"b6a26cc81faa7efafc136dd1c49eab512630a330"},{"buildStatus":{"state":"running","success":false,"hasRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/3196","number":3196,"commit":"ff8fa4c186859a0c8704ce328fc3e788e08ba5cf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/1743","number":1743,"commit":"09e70ee0ec2b631489f9acce65d2a84a0cb703aa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/878","number":878,"commit":"d83cc97980995553dcd4b23b97fd40dc7d715066"},{"buildStatus":{"state":"running","success":false,"hasRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/835","number":835,"commit":"20a9f35b98143566e13c41a6ec83ca17f9d339ab"}],"number":3278}
buildkite-pr-comment-->

angorayc committed:
remove enzyme

angorayc committed:
unit tests

stephmilovic commented on x-pack/plugins/security_solution/public/common/components/sourcerer/index.tsx:
```diff
@@ -7,105 +7,110 @@
+    selectableOptions,
+    selectedOptions,
+    setIndexPatternsByDataView,
+  } = usePickIndexPatterns({
```
love the new hook!

stephmilovic reviewed (approved):
LGTM, great work on this! The new hook makes the component so clean. Thanks for your work on this @angorayc LGTM

madirey committed:
[Security Solution][RAC][Cases] Fix RAC "add to case" functionality from alerts table (#116768)

* Fix add to case functionality

* Use appropriate owner when attaching an alert to a case

* Use field name constants

* Gotta reskip the test

* Better error handling

* Fix type errors

* Fix tests

yctercero committed:
[RAC][Timeline] - Add audit log to RBAC wrapped search strategy (#112040)

### Summary

Went back to add audit logging to the alerts table search strategy used to query RAC alerts. This PR also includes tests for the logging.

qn895 committed:
[ML] Add tests for anomaly embeddables migrations (#116520)

* [ML] Add tests for anomaly charts embeddable migrations

* [ML] Broaden tests for anomaly swimlane as well

* [ML] Fix function rename

* [ML] Update tests to use bulk api

* [ML] Remove override

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

qn895 committed:
[ML] Set ignore_throttled (#117208)

dimaanj committed:
[Discover] Fix time interval warning (#116606)

* [Discover] fix time interval warning

* [Discover] apply suggestions

* [Discover] return check

* [Discover] fix typo

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

ashokaditya committed:
Show correct completed response message on activity log (#117407)

tsullivan committed:
[Reporting] clean up config tests (#117411)

academo committed:
[Security Solution] Improves host isolation exceptions test reliability (#117327)

Commit:
[eslint] enable type-specific lint rules (#114184)

* [eslint] enable type-specific lint rules

* autofix violations

* duplicate eslint-disable to new export statement

Co-authored-by: spalger <spalger@users.noreply.github.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
fix buildkite yaml

Commit:
[docs] document using your @elastic.co email to commit (#117428)

mistic committed:
chore(NA): creates wrapper macro for ts_project rule (#117424)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

cauemarcondes committed:
[APM] Re-enable metric-based ui (#117021)

jportner committed:
Delete legacy URL aliases when objects are deleted or unshared (#117056)

yctercero committed:
[Security Solution][Platform] - Update rule exported counts to include total object count (#116338)

### Summary

Addresses #116330.

Commit:
[vscode] ignore tsconfig.tsbuildinfo and .map files (#117458)

thomheymann committed:
Fix flaky test (#117372)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

hop-dev committed:
add critical level to all fleet deprecations (#117332)

michaelolo24 committed:
[Security Solution][Investigations] - Update timeline host and ip fields to be clickable (#117403)

michaelolo24 committed:
[Security Solution][Investigations] Fix filter out of empty values (#117422)

michaelolo24 committed:
[Security Solution][Investigations] - Default enable copy icon tooltip (#117412)

dgieselaar committed:
[APM] Use synthtrace for /internal/apm/services API tests (#117083)

ashokaditya committed:
[Security Solution][Endpoint] Use a feature flag to use the new pending actions logic (#117219)

* use a feature flag to use the new pending actions logic

refs elastic/kibana/issues/116715

* switch off pending actions for endpoints when feature flag is disabled

review suggestions

* update/add tests to use FF

* correctly override the FF

`parseExperimentalConfigValue` method sets feature flag key values to `true` if passed as arg

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dimaanj committed:
[Discover] fix context loading view (#115710)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mistic committed:
skip flaky suite (#117404, 117403)

stephmilovic committed:
[Security Solution] Fix ip.replace error on Network/HTTP Tab (#116288)

mistic committed:
skip flaky suite (#117462)

jgowdyelastic committed:
[File data visualizer] Disabling create data view based on capabilities (#117347)

* [File data visualizer] Disabling create data view based on capabilities

* removing data view management card based on permissions

* fixing reset button and index data viz link

Commit:
[APM] E2E script improvements (#116972)

carlosdelest committed:
Enterprise Search - Fix event structure name change (#117502)

dominiqueclarke committed:
increase timeout for long running test (#117375)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dimaanj committed:
[Discover] Move truncate-by-height into Discover (#114320)

* [Discover] move source field formatter to discover

* [Discover] cleanup source field from field formatters

* [Discover] return source field format

* [Discover] move truncate by height to discover settings category, apply css via emotion

* [Discover] improve code readability, fix i18n

* [Discover] fix remaining i18n

* [Discover] fix unit tests

* [Discover] return truncate-by-height setting to general

* [Discover] return i18n naming

* [Discover] apply suggestions

* [Discover] fix i18n

* Update src/plugins/discover/server/ui_settings.ts

Co-authored-by: Matthias Wilhelm <ankertal@gmail.com>

* [Discover] fix embeddable and discover grid truncation styles

* [Discover] fix tests

* [Discover] get rid of emotion

* [Discover] apply suggestions

* [Discover] inject emotion styles properly

* [Discover] remove console.log

* [Discover] revert react router changes

* [Discover] fix truncate max height reset

* [Discover] simplify

* [Discover] return injection to the right place

* [Discover] remove unused import

* [Discover] apply suggestions

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Matthias Wilhelm <ankertal@gmail.com>

FrankHassanabad committed:
Increases the timeout for the test from default 1 second. (#117389)

## Summary

Fixes https://github.com/elastic/kibana/issues/115624

by increasing the timeout from 1 second to 5 seconds. Although jest is 5 seconds by default the `waitForNextUpdate` is by default 1 second. This adds a simple configuration to have it wait for 20 seconds before timing out which is what the flake is hinting at is a timeout that is beyond 1 second.

drewdaemon committed:
fix alignment (#117378)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
[APM] Improve service inventory performance (#116258)

* [APM] Improve service inventory perf

* Remove optimization for transaction groups api

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

flash1293 committed:
unskip rollup (#116567)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

cnasikas committed:
[Cases] Align cases lint rules with security solution (#117177)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mshustov committed:
remove unnecessary global typings (#117408)

* remove unnecessary global typings

* clanup index.ts

* put storybook typings back

FrankHassanabad committed:
[Security Solutions] Replaces console logging with the correct e2e logger in the e2e tests (#116872)

## Summary

This replaces the `console.log` we have with `ToolingLog` which is what the e2e tests want to use. Right now they're producing a lot of `console.log` debugging information and noise which we don't want. Instead we want to use the standard tool logging that e2e has for the FTR tests.

The two most important files with changes are:
* x-pack/test/detection_engine_api_integration/utils.ts 
* x-pack/test/lists_api_integration/utils.ts

From there everything else is just a repeating pattern of:

```ts
const log = getService('log');
```

And then pushing the log into the utilities:

such as:

```ts
await deleteListsIndex(supertest, log);
```

For reviewers, if you want to, checkout this branch and just ensure I didn't miss a `console.log` statement within the e2e tests anywhere. Also double check I did the correct `log.debug` vs `log.error` or if you think I misinterpreted one, point it out. I use `log.debug` where I have retries and then I use `log.error` where I thought it would be most useful to call out a very probable or unexpected statement happening when the tests begin to go sideways or have issues.

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

XavierM committed:
[SECURITY] Flaky api keys grid (#116991)

* fix flaky test + adding functional test

* review + remove more jest flaky test

* fix type

* fix review/confusion

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Zacqary committed:
[Metrics UI] Add docs link to redundant groupBy detection (#116822)

* [Metrics UI] Fix OR logic on redundant groupBy detection

* Switch regex change to doc change

* Fix core docs links

* Update x-pack/plugins/infra/public/alerting/metric_threshold/components/expression.tsx

Co-authored-by: Sandra G <neptunian@users.noreply.github.com>

Co-authored-by: Sandra G <neptunian@users.noreply.github.com>

ymao1 committed:
Fixing Failing test: Chrome X-Pack UI Functional Tests.x-pack/test/functional_with_es_ssl/apps/triggers_actions_ui/alert_create_flyout·ts - Actions and Triggers app create alert should successfully test valid es_query alert (#114917)

* Unskipping test

* Commenting out fix and fixing another test

* Fixing functional test

* Fixing functional test agaaaiiinn

* Fixing the actual flake

* Splitting into two tests

* Maybe this time

hop-dev committed:
[Fleet] Fix agent logs not reading query from URL (#117286)

* Don't use hash for agent logs url state

* revert core app services change

Zacqary committed:
[Logs UI][Metrics UI] Remove deprecated config fields from APIs and SavedObjects (#116821)

* [Logs UI][Metrics UI] Remove deprecated config fields from APIs

* Fix typecheck

* Fix typecheck

* Fix typecheck

* Fix jest

* Fix functional test

* Remove extraneous timeField args

* Typecheck fix

* Consolidate log file changes to ResolvedLogSourceConfiguration

* Fix merge

* Revert additional logs files

* Revert inventory models

* Revert log_analysis api

* Fix timefield reference in process list

* Restore logs page files, fix typecheck on mock

* Fix functional test

* Restore inventory models index

* Fix typecheck on getFilteredMetrics

* Look CI if you don't tell me all the type errors at once I can't fix them all

* Maybe this is the last typecheck fix who knows

* Restore reading timestamp field from data view

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

nreese committed:
[Maps] update docs for index pattern -> data view rename (#117400)

* [Maps] update docs for index pattern -> data view rename

* Update docs/maps/reverse-geocoding-tutorial.asciidoc

Co-authored-by: gchaps <33642766+gchaps@users.noreply.github.com>

Co-authored-by: gchaps <33642766+gchaps@users.noreply.github.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

thomasneirynck committed:
[Fleet] Default to APM tutorial (#117421)

claudiopro committed:
Refines hasAnyData check for alerts (#117499)

* Refines hasAnyData check for alerts

* Fixes type refinements

* Applies review feedback

* Naming is hard

dominiqueclarke committed:
adjust the synthetics journey type (#117316)

peteharverson committed:
[ML] Hide anomaly entity filter button tooltips when clicked (#117493)

* [ML] Hide anomaly entity filter button tooltips when clicked

* [ML] Move logic into blurButtonOnClick helper function

stephmilovic committed:
resolve conflicts

stephmilovic committed:
[Security Solution] [Sourcerer] [Feature Branch] Update to use Kibana Data Views  (#114806)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/4518)
* Commit: 7ab2c2aa10e210e536716df9ef44bdc36dba63df
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/4518#f7a4db0c-4f40-474b-96f2-cb88b23d99bf)

### History
* :broken_heart: [Build #3573](https://buildkite.com/elastic/kibana-pull-request/builds/3573) failed 7dd6639b3cdad7c56345b8b005f47b06565fcf44
* :broken_heart: [Build #3278](https://buildkite.com/elastic/kibana-pull-request/builds/3278) failed b6a26cc81faa7efafc136dd1c49eab512630a330
* :broken_heart: [Build #3196](https://buildkite.com/elastic/kibana-pull-request/builds/3196) failed ff8fa4c186859a0c8704ce328fc3e788e08ba5cf
* :broken_heart: [Build #1743](https://buildkite.com/elastic/kibana-pull-request/builds/1743) failed 09e70ee0ec2b631489f9acce65d2a84a0cb703aa
* :broken_heart: [Build #878](https://buildkite.com/elastic/kibana-pull-request/builds/878) failed d83cc97980995553dcd4b23b97fd40dc7d715066

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @machadoum

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/4518","number":4518,"commit":"7ab2c2aa10e210e536716df9ef44bdc36dba63df"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/3573","number":3573,"commit":"7dd6639b3cdad7c56345b8b005f47b06565fcf44"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/3278","number":3278,"commit":"b6a26cc81faa7efafc136dd1c49eab512630a330"},{"buildStatus":{"state":"running","success":false,"hasRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/3196","number":3196,"commit":"ff8fa4c186859a0c8704ce328fc3e788e08ba5cf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/1743","number":1743,"commit":"09e70ee0ec2b631489f9acce65d2a84a0cb703aa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/878","number":878,"commit":"d83cc97980995553dcd4b23b97fd40dc7d715066"}],"number":4518}
buildkite-pr-comment-->

stephmilovic committed:
Fix merge

stephmilovic commented:
this is what happens when i think i know how to git
<img width="223" alt="Screen Shot 2021-11-04 at 4 17 05 PM" src="https://user-images.githubusercontent.com/6935300/140428154-e582d3c6-6a6d-4e6f-bc14-c10b471859c6.png">
